components/lib/crash/metrics.yaml

```diff
@@ -0,0 +1,33 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# This file defines the metrics that are recorded by glean telemetry. They are
+# automatically converted to Kotlin code at build time using the `glean_parser`
+# PyPI package.
+
+$schema: moz://mozilla.org/schemas/glean/metrics/1-0-0
+
+crash_metrics:
+  uncaught_exception:
```

Comment:
What about using a single [labeled counter](https://github.com/mozilla-mobile/android-components/blob/master/components/service/glean/docs/metrics/adding-new-metrics.md#labeled-metrics) here?
With the labels `uncaught_exception`, `native_code_crash`, `other` or so.

Reply:
lib-crash only records the two types of crashes so there is no `other`.  It seemed simple enough to instrument as two plain counters rather than a single labeled one.  If we wanted to use a labeled counter for this, I would almost rather use the 'labeled' bit to record the exception that was raised rather than which type of crash it was, that seems like it would be more useful to me, but I am open to suggestions here.

Reply:
Ok, my only strong opinion here is lets stick with simple counts first.

For collecting more rich data, the question is what data in which shape is useful to the Android engineering teams, which requires a conversation we should have separately later.

A labeled counter here would match the semantics of "these two counts belong together, counting to versions of the same thing".
That matches what on desktop we use `categorical` histograms for, see e.g. [this](https://mzl.la/2KcDEKg). It has more than two labels, but this shows the concept well. You know you will only record one of them for each occurence, that you can sum them and that you can graph them together.

Reply:
Okay, you have convinced me :)